python/lsst/pipe/base/graph/graph.py

```diff
@@ -897,7 +896,7 @@ def _buildSaveObject(self, returnHeader: bool = False) -> Union[bytearray, Tuple
             taskDescription = {}
             # save the fully qualified name, as TaskDef not not require this,
             # but by doing so can save space and is easier to transport
-            taskDescription["taskName"] = f"{taskDef.taskClass.__module__}.{taskDef.taskClass.__qualname__}"
+            taskDescription["taskName"] = get_full_type_name(taskDef.taskClass)
```

Comment:
Maybe the "TaskDef not not require" should be fixed at this time?  I'm really not sure what it's trying to say.

Reply:
Does not require the fully qualified task name

Reply:
FWIW, on DM-32027, I've changed the docs to say that a fully-qualified task name is required, as there is code in various places that assumes that it is fully-qualified by calling `doImport` on it, and all of our pipeline definition files do use fully-qualified names.  Passing around a Python class name without the module and expecting magic to resolve it was always a bad idea, I think.

Reply:
Yeah I think TaskDef should have required it, and we should make it so if you didn't do it in that ticket, because it is implicitly required

Reply:
The code already used the module so was seemingly fully-qualified. I can fix the typo in the comment and look at @TallJimbo 's commit.

Reply:
It was qualified because it is the right thing to do, the comment was basically saying `TaskDef` does not require this for construction, but we are going to do it because it is better to.